packages/app/src/app.ts

```diff
@@ -338,29 +340,28 @@ export class App<Req extends Request = Request, Res extends Response = Response>
         path: '/'
       })
     }
-
     mw.push({
       handler: this.noMatchHandler,
       type: 'mw',
       path: '/'
     })
-
     const handle = (mw: Middleware) => async (req: Req, res: Res, next?: NextFunction) => {
       const { path, handler, regex } = mw
 
+      req.url = reqUrlCopy // reset req.url since it is changed in the handle fn
```

Comment:
why do we reset req.url here?

Reply:
If you have multiple middlewares for 1 path, the `req.url` is modified as per

https://github.com/tinyhttp/tinyhttp/blob/7bf29536a0e5574a82768d87a0b57a34bf217fa5/packages/app/src/app.ts#L376

and hence, every single middleware after the first receives the altered `req.url`.

Reply:
req.url is supposed to be modified. It's different for subapps. that's why `req.originalUrl` exists